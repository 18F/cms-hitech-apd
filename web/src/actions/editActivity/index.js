export * from './contractorResources';
export * from './costAllocate';
export * from './overview';
