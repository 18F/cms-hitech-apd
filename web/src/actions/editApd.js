import { updateBudget } from './apd';

export const ADD_APD_ITEM = Symbol('add apd item');
export const EDIT_APD = Symbol('edit apd');
export const REMOVE_APD_ITEM = Symbol('remove apd item');

export const addKeyPerson = () => (dispatch, getState) => {
  dispatch({
    type: ADD_APD_ITEM,
    path: '/keyPersonnel/-',
    state: getState()
  });
};

export const removeKeyPerson = (
  index,
  { global = window } = {}
) => dispatch => {
  if (global.confirm('Do you really want to delete this key person?')) {
    dispatch({
      type: REMOVE_APD_ITEM,
      path: `/keyPersonnel/${index}`
    });
    dispatch(updateBudget());
  }
};

export const setKeyPersonName = (name, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/name`,
  value: name
});

export const setKeyPersonEmail = (email, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/email`,
  value: email
});

export const setKeyPersonRole = (role, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/position`,
  value: role
});

export const setKeyPersonPercentTime = (percent, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/percentTime`,
  value: percent
});

// needs to trigger budget calculation
export const setKeyPersonHasCosts = (hasCosts, index) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/hasCosts`,
    value: hasCosts
  });
  dispatch(updateBudget());
};

// needs to trigger budget calculation
export const setKeyPersonCost = (cost, index, year) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/costs/${year}`,
    value: cost
  });
  dispatch(updateBudget());
};

export const setMedicaidDirectorEmail = email => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidDirector/email',
    value: email
  };
};

export const setMedicaidDirectorName = name => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidDirector/name',
    value: name
  };
};

export const setMedicaidDirectorPhoneNumber = phone => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidDirector/phone',
    value: phone
  };
};

export const setMedicaidOfficeAddress1 = address => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/address1',
    value: address
  };
};

export const setMedicaidOfficeAddress2 = address => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/address2',
    value: address
  };
};

export const setMedicaidOfficeCity = city => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/city',
    value: city
  };
};

export const setMedicaidOfficeState = state => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/state',
    value: state
  };
};

export const setMedicaidOfficeZip = zip => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/zip',
    value: zip
  };
};
