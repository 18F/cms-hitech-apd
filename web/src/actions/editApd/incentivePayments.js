import { EDIT_APD } from './symbols';

export const setIncentiveEHCount = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/ehCt/${year}/${quarter}`,
  value
});

export const setIncentiveEHPayment = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/ehAmt/${year}/${quarter}`,
  value
});

export const setIncentiveEPCount = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/epCt/${year}/${quarter}`,
  value
});

export const setIncentiveEPPayment = (year, quarter, value) => ({
  type: EDIT_APD,
  path: `/incentivePayments/epAmt/${year}/${quarter}`,
  value
});
