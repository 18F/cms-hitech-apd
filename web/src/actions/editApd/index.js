export * from './symbols';

export * from './keyPersonnel';
export * from './overview';
export * from './previousActivities';
export * from './stateProfile';
