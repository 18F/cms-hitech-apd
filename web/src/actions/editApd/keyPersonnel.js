import { ADD_APD_ITEM, EDIT_APD, REMOVE_APD_ITEM } from './symbols';
import { updateBudget } from '../apd';

export const addKeyPerson = () => (dispatch, getState) => {
  dispatch({
    type: ADD_APD_ITEM,
    path: '/keyPersonnel/-',
    state: getState()
  });
};

export const removeKeyPerson = (
  index,
  { global = window } = {}
) => dispatch => {
  if (global.confirm('Do you really want to delete this key person?')) {
    dispatch({
      type: REMOVE_APD_ITEM,
      path: `/keyPersonnel/${index}`
    });
    dispatch(updateBudget());
  }
};

export const setKeyPersonName = (name, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/name`,
  value: name
});

export const setKeyPersonEmail = (email, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/email`,
  value: email
});

export const setKeyPersonRole = (role, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/position`,
  value: role
});

export const setKeyPersonPercentTime = (percent, index) => ({
  type: EDIT_APD,
  path: `/keyPersonnel/${index}/percentTime`,
  value: percent
});

// needs to trigger budget calculation
export const setKeyPersonHasCosts = (hasCosts, index) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/hasCosts`,
    value: hasCosts
  });
  dispatch(updateBudget());
};

// needs to trigger budget calculation
export const setKeyPersonCost = (cost, index, year) => dispatch => {
  dispatch({
    type: EDIT_APD,
    path: `/keyPersonnel/${index}/costs/${year}`,
    value: cost
  });
  dispatch(updateBudget());
};
