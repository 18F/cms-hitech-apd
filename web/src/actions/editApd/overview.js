import { ADD_APD_YEAR, EDIT_APD, REMOVE_APD_YEAR } from './symbols';
import { updateBudget } from '../apd';

export const addYear = year => (dispatch, getState) => {
  dispatch({ type: ADD_APD_YEAR, value: year, state: getState() });
  dispatch(updateBudget());
};

export const removeYear = year => (dispatch, getState) => {
  dispatch({ type: REMOVE_APD_YEAR, value: year, state: getState() });
  dispatch(updateBudget());
};

export const setNarrativeForHIE = text => ({
  type: EDIT_APD,
  path: '/narrativeHIE',
  value: text
});

export const setNarrativeForHIT = text => ({
  type: EDIT_APD,
  path: '/narrativeHIT',
  value: text
});

export const setNarrativeForMMIS = text => ({
  type: EDIT_APD,
  path: '/narrativeMMIS',
  value: text
});

export const setProgramOverview = text => ({
  type: EDIT_APD,
  path: '/programOverview',
  value: text
});
