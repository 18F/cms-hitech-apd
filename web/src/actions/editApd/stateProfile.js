import { EDIT_APD } from './symbols';

export const setMedicaidDirectorEmail = email => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidDirector/email',
  value: email
});

export const setMedicaidDirectorName = name => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidDirector/name',
  value: name
});

export const setMedicaidDirectorPhoneNumber = phone => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidDirector/phone',
  value: phone
});

export const setMedicaidOfficeAddress1 = address => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidOffice/address1',
  value: address
});

export const setMedicaidOfficeAddress2 = address => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidOffice/address2',
  value: address
});

export const setMedicaidOfficeCity = city => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidOffice/city',
  value: city
});

export const setMedicaidOfficeState = state => ({
  type: EDIT_APD,
  path: '/stateProfile/medicaidOffice/state',
  value: state
});

export const setMedicaidOfficeZip = zip => {
  return {
    type: EDIT_APD,
    path: '/stateProfile/medicaidOffice/zip',
    value: zip
  };
};
