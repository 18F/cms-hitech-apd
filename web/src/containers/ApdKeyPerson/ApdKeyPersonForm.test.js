import { mount } from 'enzyme';
import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';

import KeyPersonForm from './ApdKeyPersonForm';
import { UPDATE_BUDGET } from '../../actions/apd';
import { EDIT_APD } from '../../actions/editApd';

const mockStore = configureStore([thunk]);

describe('the ApdKeyPersonForm component', () => {
  const props = {
    index: 1,
    item: {
      costs: {
        '1992': 100,
        '1993': 300
      },
      email: 'email address',
      hasCosts: true,
      key: 'person key',
      name: 'Bob the Builder',
      percentTime: '32',
      position: 'The Builder'
    },
    years: ['1992', '1993']
  };

  const store = mockStore('old state');

  const component = mount(
    <Provider store={store}>
      <KeyPersonForm {...props} />
    </Provider>
  );

  beforeEach(() => {
    store.clearActions();
  });

  it('renders correctly', () => {
    expect(component).toMatchSnapshot();
  });

  it('renders correctly if the person does not have costs', () => {
    expect(
      mount(
        <Provider store={store}>
          <KeyPersonForm {...props} item={{ ...props.item, hasCosts: false }} />
        </Provider>
      )
    ).toMatchSnapshot();
  });

  describe('events', () => {
    [
      ['apd-state-profile-pocname1', '/keyPersonnel/1/name'],
      ['apd-state-profile-pocemail1', '/keyPersonnel/1/email'],
      ['apd-state-profile-pocposition1', '/keyPersonnel/1/position'],
      ['apd-state-profile-pocpercentTime1', '/keyPersonnel/1/percentTime']
    ].forEach(([formName, path]) => {
      it(`handles changing the ${path}`, () => {
        component
          .findWhere(c => c.prop('name') === formName)
          .first()
          .prop('onChange')({ target: { value: 'new value' } });

        expect(store.getActions()).toEqual([
          { type: EDIT_APD, path, value: 'new value' }
        ]);
      });
    });

    it('handles toggling hasCosts off', () => {
      component
        .findWhere(
          c => c.name() === 'ChoiceComponent' && c.prop('value') === 'no'
        )
        .prop('onChange')();

      expect(store.getActions()).toEqual([
        {
          type: EDIT_APD,
          path: '/keyPersonnel/1/hasCosts',
          value: false
        },
        {
          type: UPDATE_BUDGET,
          state: 'old state'
        }
      ]);
    });

    it('handles toggling hasCosts on', () => {
      component
        .findWhere(
          c => c.name() === 'ChoiceComponent' && c.prop('value') === 'yes'
        )
        .prop('onChange')();

      expect(store.getActions()).toEqual([
        {
          type: EDIT_APD,
          path: '/keyPersonnel/1/hasCosts',
          value: true
        },
        {
          type: UPDATE_BUDGET,
          state: 'old state'
        }
      ]);
    });

    it('handles changing cost for FFY', () => {
      const hasCostsForm = mount(
        component
          .findWhere(
            c => c.name() === 'ChoiceComponent' && c.prop('value') === 'yes'
          )
          .prop('checkedChildren')
      );

      hasCostsForm
        .find('DollarField')
        .first()
        .prop('onChange')({ target: { value: 9000 } });

      expect(store.getActions()).toEqual([
        {
          type: EDIT_APD,
          path: '/keyPersonnel/1/costs/1992',
          value: 9000
        },
        {
          type: UPDATE_BUDGET,
          state: 'old state'
        }
      ]);
    });
  });
});
