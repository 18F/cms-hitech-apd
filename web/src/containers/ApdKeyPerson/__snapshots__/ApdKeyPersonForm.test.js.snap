// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the ApdKeyPersonForm component renders correctly 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <PersonForm
    index={1}
    item={
      Object {
        "costs": Object {
          "1992": 100,
          "1993": 300,
        },
        "email": "email address",
        "hasCosts": true,
        "key": "person key",
        "name": "Bob the Builder",
        "percentTime": "32",
        "position": "The Builder",
      }
    }
    years={
      Array [
        "1992",
        "1993",
      ]
    }
  >
    <h4
      className="ds-h4"
    >
      Additional
       APD Point of Contact
    </h4>
    <TextField
      autoFocus="true"
      label="Name"
      name="apd-state-profile-pocname1"
      onChange={[Function]}
      type="text"
      value="Bob the Builder"
    >
      <div
        className="ds-u-clearfix"
      >
        <FormLabel
          component="label"
          fieldId="textfield_1"
        >
          <label
            className="ds-c-label"
            htmlFor="textfield_1"
          >
            <span
              className=""
            >
              Name
            </span>
          </label>
        </FormLabel>
        <input
          autoFocus="true"
          className="ds-c-field"
          id="textfield_1"
          name="apd-state-profile-pocname1"
          onChange={[Function]}
          type="text"
          value="Bob the Builder"
        />
      </div>
    </TextField>
    <TextField
      label="Email"
      name="apd-state-profile-pocemail1"
      onChange={[Function]}
      type="text"
      value="email address"
    >
      <div
        className="ds-u-clearfix"
      >
        <FormLabel
          component="label"
          fieldId="textfield_2"
        >
          <label
            className="ds-c-label"
            htmlFor="textfield_2"
          >
            <span
              className=""
            >
              Email
            </span>
          </label>
        </FormLabel>
        <input
          className="ds-c-field"
          id="textfield_2"
          name="apd-state-profile-pocemail1"
          onChange={[Function]}
          type="text"
          value="email address"
        />
      </div>
    </TextField>
    <TextField
      label="Role"
      name="apd-state-profile-pocposition1"
      onChange={[Function]}
      type="text"
      value="The Builder"
    >
      <div
        className="ds-u-clearfix"
      >
        <FormLabel
          component="label"
          fieldId="textfield_3"
        >
          <label
            className="ds-c-label"
            htmlFor="textfield_3"
          >
            <span
              className=""
            >
              Role
            </span>
          </label>
        </FormLabel>
        <input
          className="ds-c-field"
          id="textfield_3"
          name="apd-state-profile-pocposition1"
          onChange={[Function]}
          type="text"
          value="The Builder"
        />
      </div>
    </TextField>
    <PercentField
      label="Percent commitment to program"
      name="apd-state-profile-pocpercentTime1"
      onChange={[Function]}
      size="small"
      value="32"
    >
      <NumberField
        fieldRef={
          Object {
            "current": <input
              class="ds-c-field ds-c-field--small"
              id="textfield_4"
              name="apd-state-profile-pocpercentTime1"
              type="text"
              value="32"
            />,
          }
        }
        label="Percent commitment to program"
        name="apd-state-profile-pocpercentTime1"
        onChange={[Function]}
        size="small"
        value="32"
      >
        <TextField
          fieldRef={
            Object {
              "current": <input
                class="ds-c-field ds-c-field--small"
                id="textfield_4"
                name="apd-state-profile-pocpercentTime1"
                type="text"
                value="32"
              />,
            }
          }
          label="Percent commitment to program"
          name="apd-state-profile-pocpercentTime1"
          onChange={[Function]}
          size="small"
          type="text"
          value="32"
        >
          <div
            className="ds-u-clearfix"
          >
            <FormLabel
              component="label"
              fieldId="textfield_4"
            >
              <label
                className="ds-c-label"
                htmlFor="textfield_4"
              >
                <span
                  className=""
                >
                  Percent commitment to program
                </span>
              </label>
            </FormLabel>
            <input
              className="ds-c-field ds-c-field--small"
              id="textfield_4"
              name="apd-state-profile-pocpercentTime1"
              onChange={[Function]}
              type="text"
              value="32"
            />
          </div>
        </TextField>
      </NumberField>
    </PercentField>
    <fieldset
      className="ds-c-fieldset"
    >
      <legend
        className="ds-c-label"
      >
        Does this person have costs directly attributable to this program?
      </legend>
      <ChoiceComponent
        checked={false}
        name="apd-state-profile-hascosts-no1"
        onChange={[Function]}
        type="radio"
        value="no"
      >
        <div>
          <Choice
            checked={false}
            inputPlacement="left"
            name="apd-state-profile-hascosts-no1"
            onChange={[Function]}
            type="radio"
            value="no"
          >
            <div
              aria-atomic={null}
              aria-live={null}
              aria-relevant={null}
            >
              <input
                checked={false}
                className="ds-c-choice"
                id="radio_apd-state-profile-hascosts-no1_5"
                name="apd-state-profile-hascosts-no1"
                onChange={[Function]}
                type="radio"
                value="no"
              />
              <FormLabel
                component="label"
                fieldId="radio_apd-state-profile-hascosts-no1_5"
              >
                <label
                  className="ds-c-label"
                  htmlFor="radio_apd-state-profile-hascosts-no1_5"
                >
                  <span
                    className=""
                  >
                    No
                  </span>
                </label>
              </FormLabel>
            </div>
          </Choice>
        </div>
      </ChoiceComponent>
      <ChoiceComponent
        checked={true}
        checkedChildren={
          <div
            className="ds-c-choice__checkedChild ds-l-form-row"
          >
            <div
              className="ds-l-col--auto"
            >
              <DollarField
                label="FFY 1992"
                name="apd-state-profile-costs1-fy1992"
                onBlur={null}
                onChange={[Function]}
                size="small"
                value={100}
              />
            </div>
            <div
              className="ds-l-col--auto"
            >
              <DollarField
                label="FFY 1993"
                name="apd-state-profile-costs1-fy1993"
                onBlur={null}
                onChange={[Function]}
                size="small"
                value={300}
              />
            </div>
            <div
              className="ds-l-col--auto"
            >
              <FormLabel
                component="label"
              >
                Total
              </FormLabel>
              <div
                className="form--computed-value__input-aligned"
              >
                <Dollars
                  long={false}
                >
                  400
                </Dollars>
              </div>
            </div>
          </div>
        }
        name="apd-state-profile-hascosts-yes1"
        onChange={[Function]}
        type="radio"
        value="yes"
      >
        <div>
          <Choice
            checked={true}
            checkedChildren={
              <div
                className="ds-c-choice__checkedChild ds-l-form-row"
              >
                <div
                  className="ds-l-col--auto"
                >
                  <DollarField
                    label="FFY 1992"
                    name="apd-state-profile-costs1-fy1992"
                    onBlur={null}
                    onChange={[Function]}
                    size="small"
                    value={100}
                  />
                </div>
                <div
                  className="ds-l-col--auto"
                >
                  <DollarField
                    label="FFY 1993"
                    name="apd-state-profile-costs1-fy1993"
                    onBlur={null}
                    onChange={[Function]}
                    size="small"
                    value={300}
                  />
                </div>
                <div
                  className="ds-l-col--auto"
                >
                  <FormLabel
                    component="label"
                  >
                    Total
                  </FormLabel>
                  <div
                    className="form--computed-value__input-aligned"
                  >
                    <Dollars
                      long={false}
                    >
                      400
                    </Dollars>
                  </div>
                </div>
              </div>
            }
            inputPlacement="left"
            name="apd-state-profile-hascosts-yes1"
            onChange={[Function]}
            type="radio"
            value="yes"
          >
            <div
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
            >
              <input
                checked={true}
                className="ds-c-choice"
                id="radio_apd-state-profile-hascosts-yes1_6"
                name="apd-state-profile-hascosts-yes1"
                onChange={[Function]}
                type="radio"
                value="yes"
              />
              <FormLabel
                component="label"
                fieldId="radio_apd-state-profile-hascosts-yes1_6"
              >
                <label
                  className="ds-c-label"
                  htmlFor="radio_apd-state-profile-hascosts-yes1_6"
                >
                  <span
                    className=""
                  >
                    Yes
                  </span>
                </label>
              </FormLabel>
              <div
                className="ds-c-choice__checkedChild ds-l-form-row"
              >
                <div
                  className="ds-l-col--auto"
                  key="1992"
                >
                  <DollarField
                    label="FFY 1992"
                    name="apd-state-profile-costs1-fy1992"
                    onBlur={null}
                    onChange={[Function]}
                    size="small"
                    value={100}
                  >
                    <NumberField
                      fieldRef={null}
                      label="FFY 1992"
                      mask="currency"
                      name="apd-state-profile-costs1-fy1992"
                      onBlur={[Function]}
                      onChange={[Function]}
                      size="small"
                      value="100"
                    >
                      <TextField
                        fieldRef={
                          Object {
                            "current": <input
                              aria-label="FFY 1992. Enter amount in dollars."
                              class="ds-c-field ds-c-field--currency ds-c-field--small"
                              id="textfield_7"
                              name="apd-state-profile-costs1-fy1992"
                              type="text"
                              value="100"
                            />,
                          }
                        }
                        label="FFY 1992"
                        mask="currency"
                        name="apd-state-profile-costs1-fy1992"
                        onBlur={[Function]}
                        onChange={[Function]}
                        size="small"
                        type="text"
                        value="100"
                      >
                        <div
                          className="ds-u-clearfix"
                        >
                          <FormLabel
                            component="label"
                            fieldId="textfield_7"
                          >
                            <label
                              className="ds-c-label"
                              htmlFor="textfield_7"
                            >
                              <span
                                className=""
                              >
                                FFY 1992
                              </span>
                            </label>
                          </FormLabel>
                          <div
                            className="ds-c-field-mask ds-c-field-mask--currency"
                          >
                            <div
                              className="ds-c-field__before ds-c-field__before--currency"
                            >
                              $
                            </div>
                            <Mask
                              mask="currency"
                            >
                              <input
                                aria-label="FFY 1992. Enter amount in dollars."
                                className="ds-c-field ds-c-field--currency ds-c-field--small"
                                id="textfield_7"
                                name="apd-state-profile-costs1-fy1992"
                                onBlur={[Function]}
                                onChange={[Function]}
                                type="text"
                                value="100"
                              />
                            </Mask>
                          </div>
                        </div>
                      </TextField>
                    </NumberField>
                  </DollarField>
                </div>
                <div
                  className="ds-l-col--auto"
                  key="1993"
                >
                  <DollarField
                    label="FFY 1993"
                    name="apd-state-profile-costs1-fy1993"
                    onBlur={null}
                    onChange={[Function]}
                    size="small"
                    value={300}
                  >
                    <NumberField
                      fieldRef={null}
                      label="FFY 1993"
                      mask="currency"
                      name="apd-state-profile-costs1-fy1993"
                      onBlur={[Function]}
                      onChange={[Function]}
                      size="small"
                      value="300"
                    >
                      <TextField
                        fieldRef={
                          Object {
                            "current": <input
                              aria-label="FFY 1993. Enter amount in dollars."
                              class="ds-c-field ds-c-field--currency ds-c-field--small"
                              id="textfield_8"
                              name="apd-state-profile-costs1-fy1993"
                              type="text"
                              value="300"
                            />,
                          }
                        }
                        label="FFY 1993"
                        mask="currency"
                        name="apd-state-profile-costs1-fy1993"
                        onBlur={[Function]}
                        onChange={[Function]}
                        size="small"
                        type="text"
                        value="300"
                      >
                        <div
                          className="ds-u-clearfix"
                        >
                          <FormLabel
                            component="label"
                            fieldId="textfield_8"
                          >
                            <label
                              className="ds-c-label"
                              htmlFor="textfield_8"
                            >
                              <span
                                className=""
                              >
                                FFY 1993
                              </span>
                            </label>
                          </FormLabel>
                          <div
                            className="ds-c-field-mask ds-c-field-mask--currency"
                          >
                            <div
                              className="ds-c-field__before ds-c-field__before--currency"
                            >
                              $
                            </div>
                            <Mask
                              mask="currency"
                            >
                              <input
                                aria-label="FFY 1993. Enter amount in dollars."
                                className="ds-c-field ds-c-field--currency ds-c-field--small"
                                id="textfield_8"
                                name="apd-state-profile-costs1-fy1993"
                                onBlur={[Function]}
                                onChange={[Function]}
                                type="text"
                                value="300"
                              />
                            </Mask>
                          </div>
                        </div>
                      </TextField>
                    </NumberField>
                  </DollarField>
                </div>
                <div
                  className="ds-l-col--auto"
                >
                  <FormLabel
                    component="label"
                  >
                    <label
                      className="ds-c-label"
                    >
                      <span
                        className=""
                      >
                        Total
                      </span>
                    </label>
                  </FormLabel>
                  <div
                    className="form--computed-value__input-aligned"
                  >
                    <Dollars
                      long={false}
                    >
                      <span
                        className="visibility--screen"
                      >
                        $400
                      </span>
                      <span
                        className="visibility--print"
                      >
                        $400
                      </span>
                    </Dollars>
                  </div>
                </div>
              </div>
            </div>
          </Choice>
        </div>
      </ChoiceComponent>
    </fieldset>
  </PersonForm>
</Provider>
`;

exports[`the ApdKeyPersonForm component renders correctly if the person does not have costs 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <PersonForm
    index={1}
    item={
      Object {
        "costs": Object {
          "1992": 100,
          "1993": 300,
        },
        "email": "email address",
        "hasCosts": false,
        "key": "person key",
        "name": "Bob the Builder",
        "percentTime": "32",
        "position": "The Builder",
      }
    }
    years={
      Array [
        "1992",
        "1993",
      ]
    }
  >
    <h4
      className="ds-h4"
    >
      Additional
       APD Point of Contact
    </h4>
    <TextField
      autoFocus="true"
      label="Name"
      name="apd-state-profile-pocname1"
      onChange={[Function]}
      type="text"
      value="Bob the Builder"
    >
      <div
        className="ds-u-clearfix"
      >
        <FormLabel
          component="label"
          fieldId="textfield_9"
        >
          <label
            className="ds-c-label"
            htmlFor="textfield_9"
          >
            <span
              className=""
            >
              Name
            </span>
          </label>
        </FormLabel>
        <input
          autoFocus="true"
          className="ds-c-field"
          id="textfield_9"
          name="apd-state-profile-pocname1"
          onChange={[Function]}
          type="text"
          value="Bob the Builder"
        />
      </div>
    </TextField>
    <TextField
      label="Email"
      name="apd-state-profile-pocemail1"
      onChange={[Function]}
      type="text"
      value="email address"
    >
      <div
        className="ds-u-clearfix"
      >
        <FormLabel
          component="label"
          fieldId="textfield_10"
        >
          <label
            className="ds-c-label"
            htmlFor="textfield_10"
          >
            <span
              className=""
            >
              Email
            </span>
          </label>
        </FormLabel>
        <input
          className="ds-c-field"
          id="textfield_10"
          name="apd-state-profile-pocemail1"
          onChange={[Function]}
          type="text"
          value="email address"
        />
      </div>
    </TextField>
    <TextField
      label="Role"
      name="apd-state-profile-pocposition1"
      onChange={[Function]}
      type="text"
      value="The Builder"
    >
      <div
        className="ds-u-clearfix"
      >
        <FormLabel
          component="label"
          fieldId="textfield_11"
        >
          <label
            className="ds-c-label"
            htmlFor="textfield_11"
          >
            <span
              className=""
            >
              Role
            </span>
          </label>
        </FormLabel>
        <input
          className="ds-c-field"
          id="textfield_11"
          name="apd-state-profile-pocposition1"
          onChange={[Function]}
          type="text"
          value="The Builder"
        />
      </div>
    </TextField>
    <PercentField
      label="Percent commitment to program"
      name="apd-state-profile-pocpercentTime1"
      onChange={[Function]}
      size="small"
      value="32"
    >
      <NumberField
        fieldRef={
          Object {
            "current": <input
              class="ds-c-field ds-c-field--small"
              id="textfield_12"
              name="apd-state-profile-pocpercentTime1"
              type="text"
              value="32"
            />,
          }
        }
        label="Percent commitment to program"
        name="apd-state-profile-pocpercentTime1"
        onChange={[Function]}
        size="small"
        value="32"
      >
        <TextField
          fieldRef={
            Object {
              "current": <input
                class="ds-c-field ds-c-field--small"
                id="textfield_12"
                name="apd-state-profile-pocpercentTime1"
                type="text"
                value="32"
              />,
            }
          }
          label="Percent commitment to program"
          name="apd-state-profile-pocpercentTime1"
          onChange={[Function]}
          size="small"
          type="text"
          value="32"
        >
          <div
            className="ds-u-clearfix"
          >
            <FormLabel
              component="label"
              fieldId="textfield_12"
            >
              <label
                className="ds-c-label"
                htmlFor="textfield_12"
              >
                <span
                  className=""
                >
                  Percent commitment to program
                </span>
              </label>
            </FormLabel>
            <input
              className="ds-c-field ds-c-field--small"
              id="textfield_12"
              name="apd-state-profile-pocpercentTime1"
              onChange={[Function]}
              type="text"
              value="32"
            />
          </div>
        </TextField>
      </NumberField>
    </PercentField>
    <fieldset
      className="ds-c-fieldset"
    >
      <legend
        className="ds-c-label"
      >
        Does this person have costs directly attributable to this program?
      </legend>
      <ChoiceComponent
        checked={true}
        name="apd-state-profile-hascosts-no1"
        onChange={[Function]}
        type="radio"
        value="no"
      >
        <div>
          <Choice
            checked={true}
            inputPlacement="left"
            name="apd-state-profile-hascosts-no1"
            onChange={[Function]}
            type="radio"
            value="no"
          >
            <div
              aria-atomic={null}
              aria-live={null}
              aria-relevant={null}
            >
              <input
                checked={true}
                className="ds-c-choice"
                id="radio_apd-state-profile-hascosts-no1_13"
                name="apd-state-profile-hascosts-no1"
                onChange={[Function]}
                type="radio"
                value="no"
              />
              <FormLabel
                component="label"
                fieldId="radio_apd-state-profile-hascosts-no1_13"
              >
                <label
                  className="ds-c-label"
                  htmlFor="radio_apd-state-profile-hascosts-no1_13"
                >
                  <span
                    className=""
                  >
                    No
                  </span>
                </label>
              </FormLabel>
            </div>
          </Choice>
        </div>
      </ChoiceComponent>
      <ChoiceComponent
        checked={false}
        checkedChildren={
          <div
            className="ds-c-choice__checkedChild ds-l-form-row"
          >
            <div
              className="ds-l-col--auto"
            >
              <DollarField
                label="FFY 1992"
                name="apd-state-profile-costs1-fy1992"
                onBlur={null}
                onChange={[Function]}
                size="small"
                value={100}
              />
            </div>
            <div
              className="ds-l-col--auto"
            >
              <DollarField
                label="FFY 1993"
                name="apd-state-profile-costs1-fy1993"
                onBlur={null}
                onChange={[Function]}
                size="small"
                value={300}
              />
            </div>
            <div
              className="ds-l-col--auto"
            >
              <FormLabel
                component="label"
              >
                Total
              </FormLabel>
              <div
                className="form--computed-value__input-aligned"
              >
                <Dollars
                  long={false}
                >
                  400
                </Dollars>
              </div>
            </div>
          </div>
        }
        name="apd-state-profile-hascosts-yes1"
        onChange={[Function]}
        type="radio"
        value="yes"
      >
        <div>
          <Choice
            checked={false}
            checkedChildren={
              <div
                className="ds-c-choice__checkedChild ds-l-form-row"
              >
                <div
                  className="ds-l-col--auto"
                >
                  <DollarField
                    label="FFY 1992"
                    name="apd-state-profile-costs1-fy1992"
                    onBlur={null}
                    onChange={[Function]}
                    size="small"
                    value={100}
                  />
                </div>
                <div
                  className="ds-l-col--auto"
                >
                  <DollarField
                    label="FFY 1993"
                    name="apd-state-profile-costs1-fy1993"
                    onBlur={null}
                    onChange={[Function]}
                    size="small"
                    value={300}
                  />
                </div>
                <div
                  className="ds-l-col--auto"
                >
                  <FormLabel
                    component="label"
                  >
                    Total
                  </FormLabel>
                  <div
                    className="form--computed-value__input-aligned"
                  >
                    <Dollars
                      long={false}
                    >
                      400
                    </Dollars>
                  </div>
                </div>
              </div>
            }
            inputPlacement="left"
            name="apd-state-profile-hascosts-yes1"
            onChange={[Function]}
            type="radio"
            value="yes"
          >
            <div
              aria-atomic="false"
              aria-live="polite"
              aria-relevant="additions text"
            >
              <input
                checked={false}
                className="ds-c-choice"
                id="radio_apd-state-profile-hascosts-yes1_14"
                name="apd-state-profile-hascosts-yes1"
                onChange={[Function]}
                type="radio"
                value="yes"
              />
              <FormLabel
                component="label"
                fieldId="radio_apd-state-profile-hascosts-yes1_14"
              >
                <label
                  className="ds-c-label"
                  htmlFor="radio_apd-state-profile-hascosts-yes1_14"
                >
                  <span
                    className=""
                  >
                    Yes
                  </span>
                </label>
              </FormLabel>
            </div>
          </Choice>
        </div>
      </ChoiceComponent>
    </fieldset>
  </PersonForm>
</Provider>
`;
